ui/src/components/tracks/base_slice_track.ts

```diff
@@ -314,6 +314,17 @@ export abstract class BaseSliceTrack<
     const result = await this.onInit();
     result && this.trash.use(result);
 
+    // Calc the number of rows based on the depth parameter.
+    const rowCount = assertExists(
+      (
+        await this.engine.query(`
+          SELECT
```

Comment:
WOuld it be better to do the order by depth desc limit 1 trick?

Reply:
Well spotted!